corva/application.py

```diff
@@ -0,0 +1,37 @@
+from typing import Callable, List, Optional
+
+
+def wrap_call_in_middleware(
```

Comment:
I don't think that we should copy the way FastAPI and Starlette does middleware. I know they use the call_next functionality, but that makes it so that you have interdependencies between middleware, and it's way too easy for a user to forget to properly handle the call_next value.

Can we handle this through using a list or chain, and the app itself runs all of the middleware?

Something like the pseudocode below:

```
    for middleware in middleware_list:
        context = self.run_middleware(context)

    return context
```

Reply:
I think deleting call_next will make life harder for us. But as for users, we can make them define special types of middleware which have only context as parameter, but then they will be able to run code that only precedes their main function.